### ISSUE TITLE
Errors running on R9 370 4GB

### ISSUE THREAD

**Issue by pcollinsca47** [Optiminer/OptiminerZero#3]
Running mine.sh on an R9 370 4GB card (that runs optiminer-zcash ok) it quickly errors out.
This is on Ubuntu 14.04

Logfile:
[2017-05-04 13:49:57.888] [info] Optiminer/Zero 1.1.0 (C) Optiminer 2017
[2017-05-04 13:49:57.888] [info] WatchDog: Executing "./watchdog-cmd.sh" if any GPU is stale for more than 30s.
[2017-05-04 13:49:57.888] [info] Connecting to zero.suprnova.cc:6568.
[2017-05-04 13:49:58.061] [info] Using generic kernel code.
[2017-05-04 13:49:58.065] [info] [GPU0] Device info: {"id": "0/0" "name": "Pitcairn" "platform": "AMD" "driver": "1800.11 (VM)"}
[2017-05-04 13:49:58.264] [info] Extranonce is '4e03005800000000000000005800034f'.
[2017-05-04 13:49:58.439] [info] Mining target is 01e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1e1
[2017-05-04 13:49:58.446] [info] Got new work.
[2017-05-04 13:49:58.657] [info] Connected to zero.suprnova.cc:6568.
[2017-05-04 13:50:03.062] [info] GPU0 Waiting for work from pool.
[2017-05-04 13:50:03.066] [info] GPU0 Got work, start mining.
[2017-05-04 13:50:03.360] [info] Got new work.
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 42401830
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 129761739
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 154599510
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 176292266
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 293208329
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 316342509
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 322568663
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 361496836
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 463601756
[2017-05-04 13:50:03.800] [error] Found solution index out of range: 488571277

**Comment by pcollinsca47** [comment]
Also eventually see:

[2017-05-04 14:03:25.699] [warning] [GPU0] Share submission failed: low difficulty share of 0.062499046325683594! (0 / 1)

**Comment by Optiminer** [comment]
I see. So it is producing wrong solutions. Please follow the steps at https://github.com/Optiminer/OptiminerZero/blob/master/reportbug.md and collect all the information.

**Comment by pcollinsca47** [comment]
os version: Linux 0b7e68 4.8.0-rc6-ethos3 #1 SMP Tue Sep 13 08:07:58 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

./optiminer-zero  version: Optiminer/Zero v1.1.0

fglrx driver

GPU:
[2017-05-08 15:06:05.006] [info] Optiminer/Zero 1.1.0 (C) Optiminer 2017
Platform (-c) 0, Device (-d) 0:
  Platform Vendor: AMD
  Device Name: Pitcairn
  Driver Version: 1800.11 (VM)
  Address bits: 64
  Available memory: 4204789760
  GCN Support: Gen 1
  Has device specific faster kernel: No

**Comment by pcollinsca47** [comment]
Same GPU & Miner hardware with the ssd swapped out to boot Windows 10. Under Windows optiminer-zero works fine. So something about my linux setup.

**Comment by Optiminer** [comment]
Thanks for the additional info. What fgrlx version? Do the errors happen every time you run it?

**Comment by pcollinsca47** [comment]
Yes these errors happen every time I run it. I tried it on another brand of 370 (MSI) and it also has the same errors every time. I tried it on a 4g R9 380 and it works fine. Also works on 4G R9 290.

Still trying to figure out how to get fgrlx version on ethos.
